sql/core/src/main/scala/org/apache/spark/sql/json/JacksonGenerator.scala

```diff
@@ -48,9 +50,9 @@ private[sql] object JacksonGenerator {
       case (DateType, v) => gen.writeString(v.toString)
       case (udt: UserDefinedType[_], v) => valWriter(udt.sqlType, udt.serialize(v))
 
-      case (ArrayType(ty, _), v: Seq[_]) =>
+      case (ArrayType(ty, _), v: ArrayData) =>
```

Comment:
Internal representation of Array data should be `ArrayData`, not `Seq[_]` any more.